### DISCUSSION TITLE
Apply and close functionality for column filter menus?

### DISCUSSION THREAD

**Discussion by schaffluetzel-gabriel** [primefaces/community#2907] [PrimeNG]
The enter key seems to only apply the filter, but keep the menu of the column filter open. Is there already a way to configure it so that it not only applies, but also closes the menu on pressing the enter key?

Specifically, I mean these filters:
https://primeng.org/table#filter-advanced

**Comment by qwadrox** [comment]
@schaffluetzel-gabriel Hi. There is no preconfigured way to close that dialog. [It only filters the table on enter. ](https://github.com/primefaces/primeng/blob/d6c7ffcfc5cb9a8ed355399ddae276c279836ccf/src/app/components/table/table.ts#L6019)
But you can close with escape or navigate to the apply button with the tab and hit enter.  A few more key press. 

Or you can have a custom solution:

Try it on: https://stackblitz.com/edit/ewjse1-esw6vt?file=src%2Fapp%2Ftable-filter-advanced-demo.ts

```ts
@ViewChildren(ColumnFilter) filters!: QueryList<ColumnFilter>;

onTableFilter(event: TableFilterEvent) {
  for (var filter of this.filters) {
    filter.overlayVisible = false;
  }
}
```

When the table is filtered I manually close all the filterMenu overlays. Of course, this is a very basic example but you can tweak it further to your preferences.
